
<!DOCTYPE HTML>

<html lang="en" class="_theme-default">

<head>
  <meta charset="utf-8">
  <title>urllib.error - Python 3.8 - W3cubDocs</title>
  
  <meta name="description" content=" Source code&#58; Lib&#47;urllib&#47;error.py ">
  <meta name="keywords" content="urllib, error, —, exception, classes, raised, request, -, python, python~3.8">
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="mobile-web-app-capable" content="yes">
  
  <link rel="canonical" href="http://docs.w3cub.com/python~3.8/library/urllib.error/">
  <link href="/favicon.png" rel="icon">
  <link rel="stylesheet" type="text/css" integrity="sha256-6/DH7X+2hvUPElJfGsvzm+tgIpmM9zjbYxnpsr6gR1A=" crossorigin="anonymous" href="/assets/application-ebf0c7ed7fb686f50f12525f1acbf39beb6022998cf738db6319e9b2bea04750.css">
  <script type="text/javascript" src="/assets/application-db285287b40ed28fac520fcfd75d7d874692b647b5b5e05968c741dda5de4148.js" integrity="sha256-2yhSh7QO0o+sUg/P1119h0aStke1teBZaMdB3aXeQUg=" crossorigin="anonymous"></script>
  <script src="/json/python~3.8.js"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-71174418-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>
  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
        (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-2572770204602497",
            enable_page_level_ads: true
        });
  </script>
  <script async custom-element="amp-auto-ads"
  src="https://cdn.ampproject.org/v0/amp-auto-ads-0.1.js">
</script>


</head>

<body>
	<div class="_app">
	<header class="_header">
  
  <form class="_search">
    <input type="search" class="_search-input" placeholder="Search&hellip;" autocomplete="off" autocapitalize="off" autocorrect="off" spellcheck="false" maxlength="20">
    <a class="_search-clear"></a>
    <div class="_search-tag"></div>
  </form>
  
  <a class="_home-link" href="/" ></a>
  <a class="_menu-link"></a>
  <h1 class="_logo">
    <a href="/" class="_nav-link" title="API Documentation Browser">W3cubDocs</a>
  </h1>
  
  <span class="_logo-sub-nav">/</span><span class="_logo-sub-nav"><a href="/python~3.8/" class="_nav-link" title="" style="margin-left:0;">Python 3.8</a></span>
  
  <nav class="_nav">
    <a href="https://tools.w3cub.com/?_sp=docs" target="_blank" class="_nav-link ">W3cubTools</a>
    <a href="/cheatsheets/" class="_nav-link ">Cheatsheets</a>
    <a href="/about/" class="_nav-link ">About</a>
  </nav>
</header>
	<section class="_sidebar">
		<div class="_list">
			
		</div>
	</section>
	<section class="_container ">
		<div class="_content">
			<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="6861657091"
     data-ad-format="link"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
			<div class="_page _python">
				
				
<h1 id="urllib-error-exception-classes-raised-by-urllib-request">urllib.error — Exception classes raised by urllib.request</h1> <p id="module-urllib.error"><strong>Source code:</strong> <a class="reference external" href="https://github.com/python/cpython/tree/3.8/Lib/urllib/error.py" target="_blank">Lib/urllib/error.py</a></p>  <p>The <a class="reference internal" href="#module-urllib.error" title="urllib.error: Exception classes raised by urllib.request."><code>urllib.error</code></a> module defines the exception classes for exceptions raised by <a class="reference internal" href="../urllib.request/#module-urllib.request" title="urllib.request: Extensible library for opening URLs."><code>urllib.request</code></a>. The base exception class is <a class="reference internal" href="#urllib.error.URLError" title="urllib.error.URLError"><code>URLError</code></a>.</p> <p>The following exceptions are raised by <a class="reference internal" href="#module-urllib.error" title="urllib.error: Exception classes raised by urllib.request."><code>urllib.error</code></a> as appropriate:</p> <dl class="exception"> <dt id="urllib.error.URLError">
<code>exception urllib.error.URLError</code> </dt> <dd>
<p>The handlers raise this exception (or derived exceptions) when they run into a problem. It is a subclass of <a class="reference internal" href="../exceptions/#OSError" title="OSError"><code>OSError</code></a>.</p> <dl class="attribute"> <dt id="urllib.error.URLError.reason">
<code>reason</code> </dt> <dd>
<p>The reason for this error. It can be a message string or another exception instance.</p> </dd>
</dl> <div class="versionchanged"> <p><span class="versionmodified changed">Changed in version 3.3: </span><a class="reference internal" href="#urllib.error.URLError" title="urllib.error.URLError"><code>URLError</code></a> has been made a subclass of <a class="reference internal" href="../exceptions/#OSError" title="OSError"><code>OSError</code></a> instead of <a class="reference internal" href="../exceptions/#IOError" title="IOError"><code>IOError</code></a>.</p> </div> </dd>
</dl> <dl class="exception"> <dt id="urllib.error.HTTPError">
<code>exception urllib.error.HTTPError</code> </dt> <dd>
<p>Though being an exception (a subclass of <a class="reference internal" href="#urllib.error.URLError" title="urllib.error.URLError"><code>URLError</code></a>), an <a class="reference internal" href="#urllib.error.HTTPError" title="urllib.error.HTTPError"><code>HTTPError</code></a> can also function as a non-exceptional file-like return value (the same thing that <a class="reference internal" href="../urllib.request/#urllib.request.urlopen" title="urllib.request.urlopen"><code>urlopen()</code></a> returns). This is useful when handling exotic HTTP errors, such as requests for authentication.</p> <dl class="attribute"> <dt id="urllib.error.HTTPError.code">
<code>code</code> </dt> <dd>
<p>An HTTP status code as defined in <a class="rfc reference external" href="https://tools.ietf.org/html/rfc2616.html" id="index-0" target="_blank"><strong>RFC 2616</strong></a>. This numeric value corresponds to a value found in the dictionary of codes as found in <a class="reference internal" href="http.server#http.server.BaseHTTPRequestHandler.responses" title="http.server.BaseHTTPRequestHandler.responses" target="_blank"><code>http.server.BaseHTTPRequestHandler.responses</code></a>.</p> </dd>
</dl> <dl class="attribute"> <dt id="urllib.error.HTTPError.reason">
<code>reason</code> </dt> <dd>
<p>This is usually a string explaining the reason for this error.</p> </dd>
</dl> <dl class="attribute"> <dt id="urllib.error.HTTPError.headers">
<code>headers</code> </dt> <dd>
<p>The HTTP response headers for the HTTP request that caused the <a class="reference internal" href="#urllib.error.HTTPError" title="urllib.error.HTTPError"><code>HTTPError</code></a>.</p> <div class="versionadded"> <p><span class="versionmodified added">New in version 3.4.</span></p> </div> </dd>
</dl> </dd>
</dl> <dl class="exception"> <dt id="urllib.error.ContentTooShortError">
<code>exception urllib.error.ContentTooShortError(msg, content)</code> </dt> <dd>
<p>This exception is raised when the <a class="reference internal" href="../urllib.request/#urllib.request.urlretrieve" title="urllib.request.urlretrieve"><code>urlretrieve()</code></a> function detects that the amount of the downloaded data is less than the expected amount (given by the <em>Content-Length</em> header). The <code>content</code> attribute stores the downloaded (and supposedly truncated) data.</p> </dd>
</dl>
<div class="_attribution">
  <p class="_attribution-p">
    © 2001–2020 Python Software Foundation<br>Licensed under the PSF License.<br>
    <a href="https://docs.python.org/3.8/library/urllib.error.html" class="_attribution-link" target="_blank">https://docs.python.org/3.8/library/urllib.error.html</a>
  </p>
</div>

				
			</div>
			<amp-auto-ads type="adsense"
              data-ad-client="ca-pub-2572770204602497">
</amp-auto-ads>
<ins class="adsbygoogle"
     style="display:block"
     data-ad-client="ca-pub-2572770204602497"
     data-ad-slot="1992473792"
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>
		</div>
	</section>

	</div>
	<svg style="display:none">
		<symbol id="icon-dir"><svg viewBox="0 0 20 20"><path d="M15 10c0 .3-.305.515-.305.515l-8.56 5.303c-.625.41-1.135.106-1.135-.67V4.853c0-.777.51-1.078 1.135-.67l8.56 5.305S15 9.702 15 10z"/></svg></symbol>
	  </svg>
</body>
</html>
